### PR TITLE
Add tryAdd method to headers

### PR DESCRIPTION
**Pull request by andreaTP**
Preparation for https://github.com/microsoft/kiota/pull/3401

### CODE REVIEW

File: components/abstractions/src/main/java/com/microsoft/kiota/Headers.java [hunk 1]
```diff
@@ -31,19 +31,36 @@ protected Headers(@Nonnull Headers headers) {
...
+        addImpl(key, value, true);
+    }
+
+    /**
+     * Adds a header to the current request if it was not already set
+     *
+     * @param key   the key of the header to add.
+     * @param value the value of the header to add.
+     */
+    public void tryAdd(@Nonnull final String key, @Nonnull final String value) {
```
Comment (baywet) [inline components/abstractions/src/main/java/com/microsoft/kiota/Headers.java:43]: maybe this should return a boolean to indicate whether or not the add was successful? (at least it's done this way in dotnet)

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for making the changes!
Can you add a changelog entry please (bump minor, set the date to tomorrow)
And update gradle.properties with the new version.

**Comment by andreaTP** [comment]
Sure! Done! 👍

**Review by baywet — APPROVED** [review]
Changelog change was done in #675 which is fine.

**Comment by andreaTP** [comment]
> Changelog change was done in https://github.com/microsoft/kiota-java/pull/675 which is fine.

I reverted it there as there is still something to fix for the linter, applied it here though

**Review by baywet — APPROVED** [review]

**Pull request merged**